features/TESTS/filesystem/heap_block_device/main.cpp

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+
+#include "HeapBlockDevice.h"
+#include <stdlib.h>
+
+using namespace utest::v1;
+
+/* It is not possible to build a KL25Z image with IAR including the file system if
+ * stack tracking statistics are enabled. If this is the case, build dummy
+ * tests.
+ */
+#if ! defined(TOOLCHAIN_IAR) && ! defined(TARGET_KL25Z) && ! defined(MBED_STACK_STATS_ENABLED)
```

Comment:
My proposal for this would be: would be good to split now because obviously our tools do not support this use case. But is that a long term solution ? What size requirement should be set for a test? Or is there any other way to handle this?

If there's a test that does not fit all targets (we will have definitely basic tests that should exercise API and fit to any target) and then tests that do lot of things.  Should our tools handle this as not supported (not a failure, a target cant support this test thus wont be reported as failure), but will be shown in the result. Thus a user would know x tests were not run due to a size failure (not enough RAM or FLASH).

Might be worth creating an issue for this to continue discussion there.